packages/sso-express/src/controllers.ts

```diff
@@ -88,8 +88,22 @@ export const loginController =
 
     const state = generators.random(32);
     req.session.oidcState = state;
+
+    // Code challenge and code verifier for PKCE support. If the clientSecret is set
+    // in the oidcConfig, then code challenge options will be included in the auth
+    // request session.
+    const codeVerifier = generators.codeVerifier();
+    const codeChallenge = generators.codeChallenge(codeVerifier);
+    const codeChallengeOpts = {};
+    if (options.oidcConfig.clientSecret) {
```

Comment:
Is there any reason to not have PKCE for confidential clients? It sounds like a good security feature that should always be enabled.
If this condition is removed, then the `codeChallengeOpts` object above is not needed and `code_challenge` and `code_challenge_method` can be added directly below

Reply:
Not having the guard clause will cause existing tests to fail. Would this not break existing code bases that aren't currently using confidential clients?

Reply:
Existing tests are failing because they have strict assertions on the authorizationUrl and callback parameters.

Now that I'm reading it again, this is the opposite of what is required by the common SSO realm. PKCE is required for public clients, not confidential ones (but should be enabled either way).

I don't expect any impact on existing applications